### PR TITLE
Dev clang format

### PR DESCRIPTION
**Pull request by MariusBgm**
## Subject

Run clang-format on all C and C++ files

## Description

Used the following script:
```
find SilKit/ Demos/ Utilities/ \
    -type f -iname *.cpp -o -iname *.hpp -o -iname *.c -o -iname *.ipp \
    -o -iname *.h   | while read fname;
do
    clang-format-18 -i --style=file  $fname;
done
```

[UPDATE]
Fixed clang-format version to 16, for compat reasons

## Developer checklist (address before review)

- [ ] Changelog.md updated
- [ ] Prepared update for depending repositories
- [ ] Documentation updated (public API changes only)
- [ ] API docstrings updated (public API changes only)
- [ ] Rebase &rarr; commit history clean
- [ ] Squash and merge &rarr; proper PR title

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -15,14 +15,19 @@ AlignConsecutiveAssignments: false
 AlignConsecutiveDeclarations: false
 AlignEscapedNewlines: DontAlign
 AlignOperands:   true
+
 AlignTrailingComments: false
+# > clang-format-13
+#    Kind: Leave
+#    OverEmptyLines: 2
```
Comment (VJanKraemer) [inline .clang-format:22]: Can we uncomment this? Below we also have the lambda configs which assume clang-format-13+

File: Demos/Ethernet/EthernetDemo.cpp [hunk 2]
```diff
@@ -29,10 +29,10 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. */
 #include <vector>
 
 #ifdef SILKIT_HOURGLASS
-#include "silkit/hourglass/SilKit.hpp"
-#include "silkit/hourglass/config/IParticipantConfiguration.hpp"
+#    include "silkit/hourglass/SilKit.hpp"
```
Comment (VJanKraemer) [inline Demos/Ethernet/EthernetDemo.cpp:32]: This looks funky 🤔

Comment (VJanKraemer) [reply Demos/Ethernet/EthernetDemo.cpp:32]: Maybe BeforeHash for 
https://clang.llvm.org/docs/ClangFormatStyleOptions.html#indentppdirectives
Would be a better Option?

File: SilKit/IntegrationTests/ITest_AsyncSimTask.cpp [hunk 3]
```diff
@@ -157,32 +154,32 @@ TEST(ITest_AsyncSimTask, test_async_simtask_nodeadlock)
     auto* asyncParticipant = testHarness.GetParticipant("Async");
     auto* async = testHarness.GetParticipant("Async")->GetOrCreateTimeSyncService();
 
-    sync->SetSimulationStepHandler([&syncTimeNs](auto now, std::chrono::nanoseconds /*duration*/) {
+    sync->SetSimulationStepHandler(
+        [&syncTimeNs](auto now, std::chrono::nanoseconds /*duration*/) {
         std::cout << "Sync SimTask now=" << now.count() << std::endl;
         syncTimeNs = now;
-    }, 1ms);
+    },
```
Comment (VJanKraemer) [inline SilKit/IntegrationTests/ITest_AsyncSimTask.cpp:161]: I have the feeling, that the brace indent might be switched here.

File: Demos/Ethernet/EthernetDemo.cpp (hunk 2 above)
Comment (MariusBgm) [reply Demos/Ethernet/EthernetDemo.cpp:32]: Looks both funny, I turned preprocessor indent off.

File: .clang-format (hunk 1 above)
Comment (MariusBgm) [reply .clang-format:22]: This looks horrible when turned on, and touches nearly all comments. I'll leave it as is

**Review by VJanKraemer — APPROVED** [review]
Looks coherent and consistent to me now.
👍

**Pull request merged**